homeassistant/components/sia/config_flow.py

```diff
@@ -0,0 +1,151 @@
+"""Config flow for sia integration."""
+from __future__ import annotations
+
+import logging
+
+from pysiaalarm import (
+    InvalidAccountFormatError,
+    InvalidAccountLengthError,
+    InvalidKeyFormatError,
+    InvalidKeyLengthError,
+    SIAAccount,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PORT, CONF_PROTOCOL
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_ACCOUNTS,
+    CONF_ADDITIONAL_ACCOUNTS,
+    CONF_ENCRYPTION_KEY,
+    CONF_IGNORE_TIMESTAMPS,
+    CONF_PING_INTERVAL,
+    CONF_ZONES,
+    DOMAIN,
+    TITLE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+HUB_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PORT): int,
+        vol.Optional(CONF_PROTOCOL, default="TCP"): vol.In(["TCP", "UDP"]),
+        vol.Required(CONF_ACCOUNT): str,
+        vol.Optional(CONF_ENCRYPTION_KEY): str,
+        vol.Required(CONF_PING_INTERVAL, default=1): int,
+        vol.Required(CONF_ZONES, default=1): int,
+        vol.Required(CONF_IGNORE_TIMESTAMPS, default=False): bool,
```

Comment:
`CONF_IGNORE_TIMESTAMPS` looks like something that belongs in an options flow. Unless its needed establish the connection and setup the device, we should take it out and add it to an options flow later.  We want to collect the minimal amount of data to get the device setup to minimize the burden on the user so we keep the barrier to entry as low as possible and they can get started quickly.  If this is something they can configure later, its best to ask them in an options flow only when they need it.

Reply:
that makes sense, should I remove this and just use the default for now, and then add the options flow in a subsequent PR? I also think that Zones might need to be enabled to be changed through the options flow, but there is no real default for those (mine is just 1 for now, but there are plenty of people who will set this up with other values), but since that is a alarm system config thing you can expect it to change as well even when all other values stay the same, so is it acceptable to have that in both the config and options flows?

Reply:
It is acceptable to be in both only if you need the data to set it up, but could change later.  Zones probably makes sense to be in both.

When you add the options flow, you should migrate `zones` to options.

Reply:
thanks, where should that happen, in the config flow (before setting up the entry), in the OptionFlow, or in the Hub (where the update is handled)?

Reply:
Ideally you would do it now so you don't have to migrate it later : https://github.com/home-assistant/core/pull/49912 allows creating options at creation now as if yesterday.

Reply:
perfect timing, I have the basics in place, I'll have a look!

Reply:
Just added the options flow for both, please have a look @bdraco